builtin/mainmenu/common.lua

```diff
@@ -5,6 +5,9 @@
 -- Global menu data
 menudata = {}
 
+-- Check for first run by looking for the presence of the user cache
+is_first_run = table.indexof(core.get_dir_list(core.get_cache_path()), "common.conf") == -1
```

Comment:
(this will work as long as nobody upgrades directly from <5.9.0 to 5.12.0: 6c6e48f)

Reply:
We could also check for `minetest.conf` instead; although we might forget this with #16092:

```suggestion
is_first_run = table.indexof(core.get_dir_list(core.get_user_path()), "minetest.conf") == -1
```

(I hope this is correct?)

Reply:
> (this will work as long as nobody upgrades directly from <5.9.0 to 5.12.0

which could happen a lot since debian will be moving directly from 5.6.1 to 5.11 or even something newer
https://tracker.debian.org/pkg/minetest
https://tracker.debian.org/pkg/luanti

Personally I would move this check to C++ and have it detect whether minetest.conf is non-existent or empty at startup

Reply:
Anyway it seems like I accidentally pushed the change to `minetest.conf` in [075d232](https://github.com/luanti-org/luanti/pull/16049/commits/075d23271a201ca66535ed75461e946c4d2a725c).

> Personally I would move this check to C++ and have it detect whether minetest.conf is non-existent or empty at startup

I'm not sure whether there is any difference between having this in Lua and C++.

Reply:
I moved it to C++: [f7edad0](https://github.com/luanti-org/luanti/pull/16049/commits/f7edad005119354da46376157a3fc0f6f59f73b5); I suppose this would be one less footgun then in case `minetest.conf` gets renamed or something.

Reply:
supposedly also means it now works if you pass a custom config path